tcms/testcases/templates/testcases/mutable.html

```diff
@@ -0,0 +1,152 @@
+{% extends "base.html" %}
+{% load i18n %}
+{% load static %}
+
+{% block head %}
+    {{ form.media }}
+{% endblock %}
+{% block title %}
+    {% trans "New TestCase" %}
+{% endblock %}
+
+{% block contents %}
+
+    <div class="container-fluid container-cards-pf">
+        <form class="form-horizontal" action="{% url 'testcases-new' %}" method="post">
+
+            <div class="form-group">
+                <label class="col-md-1 col-lg-1" for="id_summary">{% trans "Summary" %}</label>
+                <div class="col-md-11 col-lg-11 {% if form.summary.errors %}has-error{% endif %}">
+                    <input type="text" id="id_summary" name="summary" value="{% firstof form.summary.value "" %}" class="form-control" required>
+                    {{ form.name.errors }}
+                </div>
+            </div>
+
+            <div class="form-group">
+                <label class="col-md-1 col-lg-1" for="id_default_tester">{% trans "Default tester" %}</label>
+                <div class="col-md-3 col-lg-3">
+                    <input type="text" id="id_default_tester" name="default_tester" value="{% firstof form.default_tester.value "" %}" class="form-control">
+                </div>
+
+                <div class="col-md-1 col-lg-1">
+                    <label for="id_product">{% trans "Product" %}</label>
+                    <a href="{% url 'admin:management_product_add' %}?_popup" id="add_id_product" alt="{% trans 'add new Product' %}" title="{% trans 'add new Product' %}">+</a>
+                </div>
+                <div class="col-md-3 col-lg-3 {% if form.product.errors %}has-error{% endif %}">
+                    <select id="id_product" name="product" class="form-control selectpicker">
+                        {% for product in form.product.field.queryset %}
+                            <option value="{{ product.pk }}" {% if product.pk == form.product.value %}selected{% endif %}>
+                                {{ product.name }}
+                            </option>
+                        {% endfor %}
+                    </select>
+                    {{ form.product.errors }}
+                </div>
+
+                <div class="col-md-1 col-lg-1">
+                    <label for="id_category">{% trans "Category" %}</label>
+                    <a href="{% url 'admin:testcases_category_add' %}?_popup" id="add_id_category" alt="{% trans 'add new Category' %}" title="{% trans 'add new Category' %}">+</a>
```

Comment:
Needs a URL parameter for product which also needs to be updated when Product changes. One of the other forms had something similar if I'm not mistaken.

Reply:
We do. In `testplans/mutable.html`. The one I was banding my head in the wall two or three weeks age. Luckily, I can reuse the solution now.

Reply:
in `testplans/mutable.html` the product-related object was `version`, here it is `category`. Again, I just copied and pasted the solution, replacing the ids and the last method call from `update_version_select_from_product` to `update_category_select_from_product`, and again we can think of abstracting it and reducing duplication. Please comment on whether I do it now, or just log an issue to think about it. 

IMO having too much generic function that receive ids and selectors will bring our js code base back to the mess we are still trying to clean, so we can live with this duplication for a while

Reply:
Nope, we do not have too much generic code in this case. Categories and Versions are different things and they must be handled independently.

Yes the code is similar and this will confuse some static analysis tools but there's nowhere more to abstract and simplify.